test/extensions/filters/network/thrift_proxy/conn_state_test.cc

```diff
@@ -0,0 +1,49 @@
+#include "extensions/filters/network/thrift_proxy/conn_state.h"
+
+#include "gtest/gtest.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace NetworkFilters {
+namespace ThriftProxy {
+
+// Test behavior of nextSequenceId()
+TEST(ThriftConnectionStateTest, NextSequenceId) {
+  ThriftConnectionState cs;
+
+  EXPECT_EQ(0, cs.nextSequenceId());
+  EXPECT_EQ(1, cs.nextSequenceId());
+
+  cs.setNextSequenceIdForTest(std::numeric_limits<int32_t>::max());
```

Comment:
I guess my suggestion makes this test a bit annoying. Unless we allow to seed the sequence id in the constructor and you seed it at max - 1?

Reply:
I'll make a constructor to handle this.